packages/javascript-sdk/src/device-client/services/index.ts

```diff
@@ -44,70 +51,91 @@ export const deviceService = ({ baseUrl, realmPath }: { baseUrl: string; realmPa
-          url: `json/realms/${realm}/users/${userId}/devices/2fa/oath/${uuid}`,
-          body: { uuid, ...body },
+          url: `json/realms/${realm}/users/${userId}/devices/2fa/oath/${device.uuid}`,
+          body: device,
         }),
       }),
 
       // push device
-      getPushDevices: builder.query<PushDevicesResponse, PushDeviceQuery>({
+      getPushDevices: builder.query<GeneralResponse<PushDevice[]>, PushDeviceQuery>({
         query: ({ realm = realmPath, userId }) =>
           `/json/realms/${realm}/users/${userId}/devices/2fa/push?_queryFilter=true`,
       }),
 
-      deletePushDevice: builder.mutation<PushDevice, DeleteDeviceQuery>({
-        query: ({ realm = realmPath, userId, uuid }) => ({
-          url: `/json/realms/${realm}/users/${userId}/devices/2fa/push/${uuid}`,
+      deletePushDevice: builder.mutation<DeletedPushDevice, DeleteDeviceQuery>({
+        query: ({ realm = realmPath, userId, device }) => ({
+          url: `/json/realms/${realm}/users/${userId}/devices/2fa/push/${device.uuid}`,
           method: 'DELETE',
           body: {},
         }),
       }),
 
       // webauthn devices
-      getWebAuthnDevices: builder.query<WebAuthnDevicesResponse, WebAuthnQuery>({
+      getWebAuthnDevices: builder.query<GeneralResponse<WebAuthnDevice[]>, WebAuthnQuery>({
         query: ({ realm = realmPath, userId }) =>
           `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn?_queryFilter=true`,
       }),
       updateWebAuthnDeviceName: builder.mutation<
         UpdatedWebAuthnDevice,
-        WebAuthnQueryWithUUID & WebAuthnBody
+        WebAuthnQuery & { device: WebAuthnDevice }
       >({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn/${device.uuid}`,
           method: 'PUT',
-          body: device satisfies WebAuthnBody,
+          body: device,
         }),
       }),
       deleteWebAuthnDeviceName: builder.mutation<
-        WebAuthnDevice,
-        WebAuthnQueryWithUUID & WebAuthnBody
+        UpdatedWebAuthnDevice,
+        WebAuthnQuery & { device: UpdatedWebAuthnDevice | WebAuthnDevice }
       >({
-        query: ({ realm = realmPath, userId, ...device }) => ({
+        query: ({ realm = realmPath, userId, device }) => ({
           url: `/json/realms/${realm}/users/${userId}/devices/2fa/webauthn/${device.uuid}`,
           method: 'DELETE',
-          body: device satisfies WebAuthnBody,
+          body: device,
         }),
       }),
-      getBoundDevices: builder.mutation<DeviceResponse, BindingDeviceQuery>({
+      getBoundDevices: builder.mutation<GeneralResponse<Device[]>, GetBoundDevicesQuery>({
         query: ({ realm = realmPath, userId }) =>
           `/json/realms/${realm}/users/${userId}/devices/2fa/binding?_queryFilter=true`,
       }),
       updateBindingDeviceName: builder.mutation<Device, BindingDeviceQuery>({
```

Comment:
We're also not consistent here. Can we make sure to use `bound` and not `binding`? So, it would be `getBoundDevices`, `updateBoundDeviceName` and `deleteBoundDevice`.